android-components/components/browser/engine-gecko/src/main/java/mozilla/components/experiment/NimbusExperimentDelegate.kt

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.experiment
+
+import mozilla.components.browser.engine.gecko.GeckoNimbus
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
+import org.mozilla.geckoview.ExperimentDelegate
+import org.mozilla.geckoview.ExperimentDelegate.ExperimentException
+import org.mozilla.geckoview.ExperimentDelegate.ExperimentException.ERROR_FEATURE_NOT_FOUND
+import org.mozilla.geckoview.GeckoResult
+
+/**
+ * Default Nimbus [ExperimentDelegate] implementation to communicate with mobile Gecko and GeckoView.
+ */
+class NimbusExperimentDelegate : ExperimentDelegate {
+
+    private val logger = Logger(NimbusExperimentDelegate::javaClass.name)
+
+    /**
+     * Retrieves experiment information on the feature for use in GeckoView.
+     *
+     * @param feature Nimbus feature to retrieve information about
+     * @return a [GeckoResult] with a JSON object containing experiment information or completes exceptionally.
+     */
+    override fun onGetExperimentFeature(feature: String): GeckoResult<JSONObject> {
+        val result = GeckoResult<JSONObject>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
+        if (nimbusFeature != null) {
+            result.complete(nimbusFeature.toJSONObject())
```

Comment:
Probably outside the scope of this change, but this made me consider: what happens if the JSON structure that Nimbus provides, differs from what we expect? Do we gracefully fail where ever we call `geckoResult.accept { }`?

Reply:
Thanks, great thought! I think the consumer will have to validate and cautiously use the provided JSON at present. 

Maybe we could file a bug for a GV utility for convenience of something like `validateExperimentJson(experimentJSON: JSONObject, expectedSchema: JSONObject)` to help the consumer with this structure validation? (Or maybe a similar JSON utility exists that we could reference? I didn't see one off-hand.)